apps/app/src/views/routes/Settings/components/profile/Profile.tsx

```diff
@@ -2,28 +2,42 @@ import {
   Alert,
   AlertIcon,
   Box,
+  Button,
   Container,
   Divider,
+  Flex,
+  Modal,
+  ModalOverlay,
   SkeletonText,
+  Spacer,
   Stack,
-  Text
+  Text,
+  useDisclosure
 } from '@chakra-ui/react';
 
 import { useQuery } from '@apollo/client';
 import { graphql } from 'apps/app/src/gql';
 import { formatAddress } from 'apps/app/src/utils';
 import { useMemo } from 'react';
-import InfoGrid from '../../../components/InfoGrid';
+import InfoGrid from '../../../../components/InfoGrid';
 import { usePhoton } from '@photonhealth/react';
+import { EditProfileAction } from './ProfileEditForm';
+import { EditIcon } from '@chakra-ui/icons';
 
 const profileQuery = graphql(/* GraphQL */ `
   query MeProfileQuery {
     me {
+      id
       name {
+        first
+        last
+        middle
+        title
```

Comment:
Do we need all these fields at this level?

If I'm reading what you're trying to do correctly, what you want is a fragment, defined down on the form level, and then included up here. And then you'd use `useFragment` in the form. You can roughly copy from `InviteItem` or `UserItem`

Reply:
done, updated to a common user fragment